src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.OpenFileTracker.cs

```diff
@@ -215,30 +247,51 @@ void WatchHierarchy(IVsHierarchy hierarchyToWatch)
                 }
 
                 // We may have multiple projects with the same hierarchy, but we can use __VSHPROPID8.VSHPROPID_ActiveIntellisenseProjectContext to distinguish
+                VisualStudioProject? project = null;
                 if (ErrorHandler.Succeeded(hierarchy.GetProperty(VSConstants.VSITEMID_ROOT, (int)__VSHPROPID8.VSHPROPID_ActiveIntellisenseProjectContext, out var contextProjectNameObject)))
                 {
                     WatchHierarchy(hierarchy);
 
                     if (contextProjectNameObject is string contextProjectName)
                     {
-                        var project = _workspace.GetProjectWithHierarchyAndName_NoLock(hierarchy, contextProjectName);
-
-                        if (project != null && projectIds.Contains(project.Id))
-                        {
-                            return project.Id;
-                        }
+                        project = _workspace.GetProjectWithHierarchyAndName(hierarchy, contextProjectName);
```

Comment:
hrmm.. this is still problematic.  we still need a lock here.  can't think of a way to avoid thsi.